src/openrct2/actions/BannerPlaceAction.cpp

```diff
@@ -86,17 +104,10 @@ GameActions::Result::Ptr BannerPlaceAction::Query() const
         return MakeResult(GameActions::Status::ItemAlreadyPlaced, STR_CANT_POSITION_THIS_HERE, STR_BANNER_SIGN_IN_THE_WAY);
     }
 
-    if (_bannerIndex == BANNER_INDEX_NULL || _bannerIndex >= MAX_BANNERS)
+    if (HasReachedBannerLimit())
     {
-        log_error("Invalid banner index, bannerIndex = %u", _bannerIndex);
-        return MakeResult(GameActions::Status::InvalidParameters, STR_CANT_POSITION_THIS_HERE);
-    }
-
-    auto banner = GetBanner(_bannerIndex);
-    if (!banner->IsNull())
-    {
-        log_error("Banner index in use, bannerIndex = %u", _bannerIndex);
-        return MakeResult(GameActions::Status::InvalidParameters, STR_CANT_POSITION_THIS_HERE);
+        log_error("No free banners available");
+        return MakeResult(GameActions::Status::InvalidParameters, STR_TOO_MANY_BANNERS_IN_GAME);
```

Comment:
I think this will result in a half-message. (I will check it.)

```suggestion
        return MakeResult(GameActions::Status::InvalidParameters, STR_CANT_POSITION_THIS_HERE, STR_TOO_MANY_BANNERS_IN_GAME);
```

Reply:
Uses the `BannerPlaceActionResult` constructor so will insert the STR_CANT_POSITION_THIS_HERE.